### ISSUE TITLE
False-positives and improvements for SimpleAccessorNameNotationCheck

### ISSUE THREAD

**Issue by daniilyar** [sevntu-checkstyle/sevntu.checkstyle#166]
1). SimpleAccessorNameNotationCheck should not display warnings if getter returns a constant. Example:

``` java
private static final ArrayList<String> ALL_ACTION_FEATURES = Lists.newArrayList();
    static {
        ALL_ACTION_FEATURES.addAll(COMPANY_RELATED_TOOL_ACTION_FEATURES);
        ALL_ACTION_FEATURES.addAll(HIER_RELATED_ACTION_FEATURES);
        ALL_ACTION_FEATURES.addAll(RISC_RELATED_ACTION_FEATURES);
    }

    public static List<String> getAllActionFeatures() {
        return ALL_ACTION_FEATURES; // Should be no warning
    }
```

2). This check should NOT show warnings on annotations:

``` java
    @SomeAnnotation // < warning is shown here instead of the next line
    public Date getUserExpirationDate() {
        return userExpDate;
    }
```

3). Check message should be updated from '_Unexpected getter/setter name._' to something like:
 _Getter/setter name does not match the name of [appropriate] field. Expected getter/setter name is <xxyyZZZ>_ to make it convenient in usage. Additional minor fix to check message: it is not necessary to have a dot at the message end.
4). Check should not warn on methods with @Override annotation.
5). We need to update JavaDoc to let user know what we count as getter/setter.

**Comment by romani** [comment]
Nikita does not have time to continue work, 

some work is done at https://github.com/GraninNikita/sevntu.checkstyle/commit/0019b772a6eab820c9dbc3d363ce243669b7ab23 please use it if that reasonabale.

**Comment by mkordas** [comment]
+1 - this issue must be resolved to make thich check usable
